src/components/SignerPanel/SignerPanel.js

```diff
@@ -0,0 +1,245 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { SidePanel, springs } from '@aragon/ui'
+import { Transition, animated } from 'react-spring'
+import { addressesEqual } from '../../web3-utils'
+import ConfirmTransaction from './ConfirmTransaction'
+import SigningStatus from './SigningStatus'
+import {
+  STATUS_CONFIRMING,
+  STATUS_SIGNING,
+  STATUS_SIGNED,
+  STATUS_ERROR,
+} from './signer-statuses'
+
+const INITIAL_STATE = {
+  panelOpened: false,
+  intent: {},
+  directPath: false,
+  actionPaths: [],
+  pretransaction: null,
+  status: STATUS_CONFIRMING,
+  signError: null,
+}
+
+class SignerPanel extends React.Component {
+  static propTypes = {
+    apps: PropTypes.array,
+    account: PropTypes.string,
+    web3: PropTypes.object,
+    transactionBag: PropTypes.object,
+  }
+
+  static defaultProps = {
+    apps: [],
+    account: '',
+  }
+
+  state = { ...INITIAL_STATE }
+
+  componentWillReceiveProps({ transactionBag }) {
+    // Received a new transaction to sign
+    if (transactionBag && transactionBag !== this.props.transactionBag) {
```

Comment:
Wondering if we should add the check for the network type here, and we can add an error type.